### DISCUSSION TITLE
Sync data between two phones

### DISCUSSION THREAD

**Discussion by Saturo78** [TWiStErRob/net.twisterrob.inventory#427] [Questions]
Hello, we would like to use the app to manage our move. I still live in the old house and am packing our things into boxes. My wife already lives in our new house and is unpacking the boxes. We are now looking for a way to keep the data on our phones synchronized.
Would it work to import each other's backups? Presumably each item would only be processed by one person, so there shouldn't actually be any conflicts?
It would be really cool if there was a sync function that stored the data on Dropbox, for example.

**Comment by TWiStErRob** [comment]
I know, this is one of the highly coveted features:

 * https://github.com/TWiStErRob/net.twisterrob.inventory/issues/53

Sharing backups work, but only if you use a primary+clone system, because the import functionality merges backups when they're imported.

So if your inventory has (primary):

 * a
   * b
 * c
   * d

then you share it, and your wife makes changes (other primary), backs up and shares it back:

 * a
   * b
   * e (added a new item)
 * c
 * f (added a new item)
   * d (moved from c)

when you import it back (into primary), the result will be:

 * a (existing item, unchanged, skipped on import)
   * b (existing item, unchanged, skipped on import)
   * e (added a new item)
 * c (existing item, unchanged, skipped on import)
   * d (PROBLEM! existing item is still in place)
 * f (added a new item)
   * d (PROBLEM! moved item shows as new, duplicating original, even worse if description/image changed too)

This means that the "other primary" has to be read only ("clone") to prevent weird data.

There's a way to do an initial inventory in parallel, by working on separate rooms/boxes, and then importing them together, but modifications/moves (which I assume will happen a lot during unpacking) are hard to merge.

---

Note: by hard I mean you can create partial backup by:

 * make a backup (to save your data)
 * delete data you don't want shared (e.g. the old or new property)
 * make a backup (to be shared)
 * delete everything
 * import original backup

On the import side, you would delete a property which is in the backup, and import the "new version" of it.
You really have to nail what to delete, otherwise you end up with the above problem or lose data (because of accidental deletion).

**Reply by Saturo78** [reply]
oh, that sounds difficult. I was hoping that each item would have a unique id that could be used to update it when it was moved.
I'll have a think about the best way to do that.
